dev/reference/spark_commands.md

```diff
@@ -3,273 +3,215 @@
 
 ---
 ### Delete Object
 
-* Opcode: 4
-* Request:
-    * Opcode: Byte
-    * ObjectId: Variable Length Id
-* Response:
-    * ErrorCode: Byte
+- Request:
+    - Opcode: `byte`
+    - ObjectId: `uint8_t`
+- Response:
+    - Errorcode: `byte`
 
 ---
-### List Objects
+### Activate Profiles
 
-* Opcode: 5
-* Request:
-    * Opcode: Byte
-    * ProfileId: int8_t
-* Response:
-    * ErrorCode: Byte
-    * Padding: Byte
-    * Objects (repeated):
-        * ObjectId: Variable Length Id
-        * ObjectType: uint16_t
-        * ObjectSize: Byte
-        * ObjectData: Byte[]
-    * Padding: Byte
-    * Terminated: Null Byte
-
-**TODO**:
-* Remove padding from response
-* Remove ObjectSize from response
+- Request:
+    - Opcode: `byte`
+    - Profiles: `bit[8]`
+- Response:
+    - Errorcode: `byte`
 
 ---
-### Free Object Slot
+### Read System Object
 
-* Opcode: 6
-* Request:
-    * Opcode: Byte
-    * ObjectId: Variable Length Id
-* Response:
-    * ErrorCode: Byte
+- Request:
+    - Opcode: `byte`
+    - ObjectId: `uint8_t`
+    - ObjectType: `uint16_t`
+- Response:
+    - Errorcode: `byte`
+    - ObjectType: `uint16_t`
+    - ObjectData: `byte[]`
 
 ---
-### Create Profile
-
-* Opcode: 7
-* Request:
-    * Opcode: Byte
-* Response:
-    * ErrorCode: Byte
-    * ProfileId: int8_t
+### Write System Object
+
+- Request:
+    - Opcode: `byte`
+    - ObjectId: `uint8_t`
+    - ObjectType: `uint16_t`
+    - ObjectData: `byte[]`
+- Response:
+    - Errorcode: `byte`
+    - ObjectType: `uint16_t`
+    - ObjectData: `byte[]`
 
 ---
-### Delete Profile
+### List Active Profiles
 
-* Opcode: 8
-* Request:
-    * Opcode: Byte
-    * ProfileId: int8_t
-* Response:
-    * ErrorCode: Byte
+- Request:
+    - Opcode: `byte`
+- Response:
+    - Errorcode: `byte`
+    - Profiles: `bit[8]`
 
 ---
-### Activate Profile
-
-* Opcode: 9
-* Request:
-    * Opcode: Byte
-    * ProfileId: int8_t
-* Response:
-    * ErrorCode: Byte
+### List Active Objects
+
+- Request:
+    - Opcode: `byte`
+- Response:
+    - Errorcode: `byte`
+    - Profiles: `bit[8]`
+- Values:
+    - ObjectId: `uint8_t`
+    - Profiles: `bit[8]`
+    - ObjectType: `uint16_t`
+    - ObjectData: `byte[]`
 
 ---
-### Log Values
-
-* Opcode: 10
-* Request:
-    * Opcode: Byte
-    * Flags: Byte with bit flags
-        * bit 1: Id chain
-        * bit 2: Log system container
-    * Optional: ObjectId: Variable Length Id
-* Response:
-    * ErrorCode: Byte
-    * Optional: Objects (repeated):
-        * ObjectId: Variable Length Id
-        * ObjectType: uint16_t
-        * ObjectSize: Byte
-        * ObjectData: Byte[]
-    * Padding: Byte
-    * Terminated: Null Byte
-
-**TODO**:
-* Remove padding from response
-* Remove ObjectSize from response
+### List Profile Objects
+
+- Request:
+    - Opcode: `byte`
+    - Profile: `uint8_t`
+- Response:
+    - Errorcode: `byte`
+- Values:
+    - ObjectId: `uint8_t`
+    - Profiles: `bit[8]`
+    - ObjectType: `uint16_t`
+    - ObjectData: `byte[]`
```

Comment:
proposal:

Change to 2 endpoints:

active objects and saved objects.
active objects returns runtime objects, saved objects returns EEPROM definitions.

These should be treated as different data. The persisted part of the live data likely matches the EEPROM data, but we don't assume this. For example and object that persists every 30 minutes.

We can return all profiles for the saved data, so this can be implemented as a get, not post.